### DISCUSSION TITLE
How can I push() a different key/value set into an object repeatedly?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#55238] [Help] [answered]
### Summary

***** Hint – I think this might be a problem of my Visual Studio Code… but you guys can read it all to see…

Problem:
I have a JSON file to which I want to add another key/value pair kind of like this:

`“text2”: “here is something else”`

Here is a sample of the original json without the addition:
```
{
“data”: [
{
“id”: “1700148573403304137”,
“text”: “hola”
},
{
“id”: “1700147255322259501”,
“text”: “hello”
},
{
“id”: “1700146984903155986”,
“text”: “hi”
}
]
}
```

Here is the code:
```
var data = fs.readFileSync(“./thejsons/a.json”);
var myObject = JSON.parse(data);

let toAdd = { text2: { valuefortext2 } };

console.log(toAdd);

myObject.data[textIndex++].push(toAdd);
var newData2 = JSON.stringify(myObject);

timesRun += 1;

if (timesRun === 3) {
  clearInterval(myIntervals);

  fs.writeFile("./thejsons/a1.json", newData2, (err) => {
    // error checking
    if (err) throw err;

    console.log("New data added");
  });

```

`Error:  is not a function`    when I try this ` myObject.data[textIndex++].push(toAdd);`

OR.... if I delete` [textIndex++]`  the `toAdd`  gets added, but just once (last data at the run), at the bottom of the .data array as a new object in the array..

I think the problem might be from my Visual Studio Code changing this format below??

`let toAdd = { text2: { valuefortext2 } };`  <--   that is what visual studio code is re-formatting this line after I save.

`let toAdd = "text2" : `${ valuefortext2 }` ;`  <---  but I want something more like this, to be added under each one of the "text" key/value pairs of the .data array...

Any feedback on how I could fix this?.....  any information would be awesome...

**Comment by Elindo586** [comment, marked as answer]
The above code wont work because you cannot push anything into a non array....

**Comment by icyJoseph** [comment]
If you want to extend the data object by pushing into it:

```js
myObject.data.push(toAdd); // note that here toAdd has a `text2` property, but you probably want `text` 
```

If you want to target a particular index of the data object and replace the text property:

```js
myObject.data[textIndex++].text = toAdd.text2
```
